assets/aml-benchmark/scripts/custom_inference_postprocessors/humaneval.py

```diff
@@ -184,17 +184,24 @@ def run_humaneval_postprocessor(
     # Post processing the prediction and ground truth columns
     for row in pred_dict_full:
         gt = "\n" + row["test"] + "\n" + "check(" + row["entry_point"] + ")"
-        if str("def " + row["entry_point"] + "(") in row["original_prediction"]:
-            # If the model regenerates the prompt/ function name
-            pred_combined_prompt = _extract_text_from_markdown_tag(row["original_prediction"], tag_type='python')
-        else:
-            original_prediction = _extract_text_from_markdown_tag(row["original_prediction"], tag_type='python')
+        tag_type = "python" if "python" in row["original_prediction"] else ""
+
+        # Extract the prediction data from the markdown tags
+        pred_combined_prompt = _extract_text_from_markdown_tag(row["original_prediction"], tag_type)
+
+        # Get the index of the first function definition and return keyword
+        func_name_index = pred_combined_prompt.find(str("def " + row["entry_point"] + "("))
+        return_keyword_index = pred_combined_prompt.find("return")
+
+        # If function definition is not present or present after the initial function body prediction
+        if func_name_index == -1 or return_keyword_index < func_name_index:
```

Comment:
[HumanEval_38.txt](https://github.com/Azure/azureml-assets/files/15020678/HumanEval_38.txt)
Handled the case.

Reply:
In default scenario, it should work even if the full text is returned. Still we are checking if the function name is present in the prediction, then we remove the function name and body from the input prompt. At max in this case the imports might be twice in case present. 
[humaneval_14.txt](https://github.com/Azure/azureml-assets/files/15020917/humaneval_14.txt)

Reply:
Will this work if the model predicted `decode_cyclic` instead of `decode_cyclic2` and `encode_cyclic` instead of `encode_cyclic2` and `encode_cyclic3`?

Reply:
Yes, it is working. 
[HumanEval_38_samefunctionnames.txt](https://github.com/Azure/azureml-assets/files/15022183/HumanEval_38_samefunctionnames.txt)